frontend/frontend.php

```diff
@@ -179,6 +184,60 @@ public function parse_tax_query( $query ) {
 		}
 	}
 
+	/**
+	 * Prevents unnecessary queries to the database by transforming
+	 * a language query by `slug` to a language query by `term_taxonomy_id`.
+	 *
+	 * These extra queries occur when the language slug is displayed in the current's page URL, because WP wants a list
+	 * of `term_taxonomy_id`.
+	 *
+	 * Example:
+	 * From: array(
+	 *     'taxonomy'         => 'language',
+	 *     'terms'            => array( 'en' ),
+	 *     'field'            => 'slug',
+	 *     'operator'         => 'IN',
+	 *     'include_children' => 1,
+	 * )
+	 * To: array(
+	 *     'taxonomy'         => 'language',
+	 *     'terms'            => array( 238 ),
+	 *     'field'            => 'term_taxonomy_id',
+	 *     'operator'         => 'IN',
+	 *     'include_children' => 1,
+	 * )
+	 *
+	 * @since 3.8
+	 *
+	 * @param WP_Query $query WP_Query object.
+	 * @return void
+	 */
+	public function transform_query( $query ): void {
```

Comment:
Also what do you think to move this to `PLL_Query` which is supposed to include our methods related to `WP_Query`?

Reply:
Why not. Done in [3385903](https://github.com/polylang/polylang/pull/1650/commits/338590358f861a365f0b500343f550c38bb48d15).